devolutions-crypto-wayk/src/bastion/key_exchange.rs

```diff
@@ -0,0 +1,206 @@
+use crate::bastion::Error;
+use uuid::Uuid;
+use x25519_dalek::EphemeralSecret;
+
+pub use x25519_dalek::{PublicKey, SharedSecret};
+
+/// Basically wraps `x25519_dalek`
+pub struct KeyExchange {
+    /// Send this to peer.
+    pub public_key: PublicKey,
+
+    /// Hidden: this should not be shared.
+    /// Good to know: this type is `Zeroize` on `Drop`.
+    secret: EphemeralSecret,
+}
+
+impl KeyExchange {
+    pub fn init() -> Self {
+        use rand_core::OsRng;
+
+        let secret = EphemeralSecret::new(OsRng);
+        Self {
+            public_key: PublicKey::from(&secret),
+            secret,
+        }
+    }
+
+    pub fn into_shared_secret(self, peer_public_key: &PublicKey) -> SharedSecret {
+        self.secret.diffie_hellman(peer_public_key)
+    }
+}
+
+/// Encrypts symmetric key to be sent over the wire using shared secret.
+///
+/// - Output is the encrypted key encoded in base64.
+///
+/// # Interals
+///
+/// - AAD is the CSC UUID in big-endian binary format.
+/// - A 24-byte initialization vector (IV) is randomly generated.
+/// - Symmetric key (our plaintext) is encrypted using XChaCha2020-Poly1035.
+/// - Output is a buffer such as `[AAD (16) | IV (24) | Ciphertext (variable) | Tag (16)]`
+///     encoded in standard base64.
+pub fn encrypt_key(
```

Comment:
Can't we simply dump and use the SharedSecret for the session key instead of adding a key wrapping? @awakecoding